### PR TITLE
Have build_and_upload_dependencies.sh use the new docker image hashes

### PR DESCRIPTION
**Pull request by beckerhe**
Docker images are now referenced by digests - basically upload hashes. This allow us to pin a specific version of a Docker image to each commit in the repo.

This change broke the `build_and_upload_dependencies.sh` script, so this PR cleans up this script, improves it, adds support for Docker digests and adds support for a potential public build of Orbit dependencies.

### CODE REVIEW

File: third_party/conan/docker/utils.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+function find_container_for_conan_profile {
+    readonly CONAN_PROFILE="$1"
+
+    # Use a specific tag from the mapping. If none is specified, fall back to `latest`.
+    readonly DOCKER_IMAGE_TAG="${docker_image_tag_mapping[${CONAN_PROFILE}]-latest}"
+
+    CONTAINER="gcr.io/orbitprofiler/${CONAN_PROFILE}:${DOCKER_IMAGE_TAG}"
+
+    if [ ${docker_image_digest_mapping[${CONAN_PROFILE}]+abc} ]; then
```
Comment (danielfenner) [inline third_party/conan/docker/utils.sh:18]: Looks like there is a digest for every profile. Why does this not completely replace the docker_image_tag_mapping and the code above?

Comment (beckerhe) [reply third_party/conan/docker/utils.sh:18]: Can probably be removed now. This mechanism was designed as having fallbacks. But yes, I can probably remove that now. I just thinking whether this has any disadvantages.

Comment (beckerhe) [reply third_party/conan/docker/utils.sh]: Fixed!

**Comment by beckerhe** [comment]
@danielfenner I'm finally coming back to this. Could you have another look?

**Review by danielfenner — APPROVED** [review]
lgtm!

**Pull request merged**